frame/state-trie-migration/src/lib.rs

```diff
@@ -0,0 +1,1417 @@
+		/// reading a key, we simply cannot know how many bytes it is. In other words, this should
+		/// not be used in any environment where resources are strictly bounded (e.g. a parachain),
+		/// but it is acceptable otherwise (relay chain, offchain workers).
+		pub(crate) fn migrate_until_exhaustion(&mut self, limits: MigrationLimits) {
+			log!(debug, "running migrations on top of {:?} until {:?}", self, limits);
+
+			if limits.item.is_zero() || limits.size.is_zero() {
+				// handle this minor edge case, else we would call `migrate_tick` at least once.
+				log!(warn, "limits are zero. stopping");
+				return
+			}
+
+			loop {
+				self.migrate_tick();
+				if self.exhausted(limits) {
+					break
+				}
+			}
+
+			// accumulate dynamic data into the storage items.
+			self.size = self.size.saturating_add(self.dyn_size);
+			self.child_items = self.child_items.saturating_add(self.dyn_child_items);
+			self.top_items = self.top_items.saturating_add(self.dyn_top_items);
+			log!(debug, "finished with {:?}", self);
+		}
+
+		/// Migrate AT MOST ONE KEY. This can be either a top or a child key.
+		///
+		/// This function is the core of this entire pallet.
+		fn migrate_tick(&mut self) {
+			match (self.current_top.as_ref(), self.current_child.as_ref()) {
+				(Some(_), Some(_)) => {
+					// we're in the middle of doing work on a child tree.
+					self.migrate_child();
+				},
+				(Some(ref top_key), None) => {
+					// we have a top key and no child key. 3 possibilities exist:
+					// 1. we continue the top key migrations.
+					// 2. this is the root of a child key, and we start processing child keys (and
+					// should call `migrate_child`).
+					// 3. this is the root of a child key, and we are finishing all child-keys (and
+					// should call `migrate_top`).
+
+					// NOTE: this block is written intentionally to verbosely for easy of
+					// verification.
+					match (
+						top_key.starts_with(DEFAULT_CHILD_STORAGE_KEY_PREFIX),
+						self.prev_tick_child,
+					) {
+						(false, false) => {
+							// continue the top key migration
+							self.migrate_top();
+						},
+						(true, false) => {
+							// start going into a child key. In the first iteration, we always
+							let maybe_first_child_key = {
+								// just in case there's some data in `&[]`, read it. Since we can't
+								// check this without reading the actual key, and given that this
+								// function should always read at most one key, we return after
+								// this. The rest of the migration should happen in the next tick.
+								let child_top_key = Pallet::<T>::child_io_key_or_halt(top_key);
+								let _ = sp_io::default_child_storage::get(child_top_key, &vec![]);
+								sp_io::default_child_storage::next_key(child_top_key, &vec![])
+							};
+							if let Some(first_child_key) = maybe_first_child_key {
+								self.current_child = Some(first_child_key);
+								self.prev_tick_child = true;
```

Comment:
You have this set in both if branches.

Reply:
indeed and it is intentional, since in my opinion the code is more readable this way, because:

1. A comment (`set the `self.current_child`, next`) and the instruction that corresponding to it (`self.prev_tick_child = true`) are placed next to each other. 
2. an `else` branch with no instructions and just comment is not desirable. 

All in all, this is an example of code being intentionally written verbosely for the sake of readability.